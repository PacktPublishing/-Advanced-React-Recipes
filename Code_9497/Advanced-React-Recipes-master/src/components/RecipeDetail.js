import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import { compose, branch, renderComponent } from 'recompose';

const LoadingState = ({ style, className }) => (
  <p
    style={style}
    className={classNames('h3 p2 bg-white italic center', className)}
  >
    Loading recipe...
  </p>
);

const ZeroState = ({ style, className }) => (
  <p
    style={style}
    className={classNames('h3 p2 bg-white italic center', className)}
  >
    No recipe selected.
  </p>
);

const RecipeDetail = ({ recipe, className, style }) => (
  <div style={style} className={classNames('p2 bg-white', className)}>
    <h2 className="h2">{recipe.name}</h2>
    <div className="flex flex-column">
      <img alt={recipe.name} className="fit" src={recipe.image} />
      <div>
        <span>{recipe.category}</span>
        <span>{recipe.calories} cal</span>
      </div>
      {recipe.ingredients && (
        <div>
          <h3>Ingredients</h3>
          <ul>
            {recipe.ingredients.map(ingredient => (
              <li key={ingredient}>{ingredient}</li>
            ))}
          </ul>
        </div>
      )}
      {recipe.steps && (
        <div>
          <h3>Steps</h3>
          <ol>{recipe.steps.map(step => <li key={step}>{step}</li>)}</ol>
        </div>
      )}
      <Link
        className="right-align"
        to={`/recipe/${recipe.id}`}
        href={`/recipe/${recipe.id}`}
      >
        Detail
      </Link>
    </div>
  </div>
);

RecipeDetail.propTypes = {
  recipe: PropTypes.object,
  className: PropTypes.string,
  style: PropTypes.object,
};

LoadingState.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
};

ZeroState.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
};

export default compose(
  branch(props => props.loading, renderComponent(LoadingState)),
  branch(props => !props.recipe, renderComponent(ZeroState)),
)(RecipeDetail);
