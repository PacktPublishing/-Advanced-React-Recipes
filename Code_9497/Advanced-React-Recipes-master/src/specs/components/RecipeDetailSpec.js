import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import RecipeDetail from '../../components/RecipeDetail';

describe('<RecipeDetail />', () => {
  let component;
  let tree;
  let testRecipeDetail;

  beforeEach(() => {
    testRecipeDetail = {
      id: 1,
      name: 'Test recipe',
      category: 'Dessert',
      calories: 100,
      image: 'test url',
      ingredients: ['Ingrendient 1', 'Ingrendient 2'],
      steps: ['Steps 1', 'Steps 2'],
    };
  });

  const recipe = (props = {}) => (
    <BrowserRouter>
      <RecipeDetail {...props} />
    </BrowserRouter>
  );

  test('Should render zero state if no recipe passed', () => {
    component = renderer.create(recipe());
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render passed recipe', () => {
    component = renderer.create(recipe({ recipe: testRecipeDetail }));
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should not render ingredients if none', () => {
    delete testRecipeDetail.ingredients;
    component = renderer.create(recipe({ recipe: testRecipeDetail }));
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should not render steps if none', () => {
    delete testRecipeDetail.steps;
    component = renderer.create(recipe({ recipe: testRecipeDetail }));
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
