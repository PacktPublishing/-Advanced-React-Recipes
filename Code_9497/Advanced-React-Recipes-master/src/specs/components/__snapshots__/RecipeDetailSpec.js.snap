// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetail /> Should not render ingredients if none 1`] = `
<div
  className="p2 bg-white"
  style={undefined}
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <div
    className="flex flex-column"
  >
    <img
      alt="Test recipe"
      className="fit"
      src="test url"
    />
    <div>
      <span>
        Dessert
      </span>
      <span>
        100
         cal
      </span>
    </div>
    <div>
      <h3>
        Steps
      </h3>
      <ol>
        <li>
          Steps 1
        </li>
        <li>
          Steps 2
        </li>
      </ol>
    </div>
    <a
      className="right-align"
      href="/recipe/1"
      onClick={[Function]}
    >
      Detail
    </a>
  </div>
</div>
`;

exports[`<RecipeDetail /> Should not render steps if none 1`] = `
<div
  className="p2 bg-white"
  style={undefined}
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <div
    className="flex flex-column"
  >
    <img
      alt="Test recipe"
      className="fit"
      src="test url"
    />
    <div>
      <span>
        Dessert
      </span>
      <span>
        100
         cal
      </span>
    </div>
    <div>
      <h3>
        Ingredients
      </h3>
      <ul>
        <li>
          Ingrendient 1
        </li>
        <li>
          Ingrendient 2
        </li>
      </ul>
    </div>
    <a
      className="right-align"
      href="/recipe/1"
      onClick={[Function]}
    >
      Detail
    </a>
  </div>
</div>
`;

exports[`<RecipeDetail /> Should render passed recipe 1`] = `
<div
  className="p2 bg-white"
  style={undefined}
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <div
    className="flex flex-column"
  >
    <img
      alt="Test recipe"
      className="fit"
      src="test url"
    />
    <div>
      <span>
        Dessert
      </span>
      <span>
        100
         cal
      </span>
    </div>
    <div>
      <h3>
        Ingredients
      </h3>
      <ul>
        <li>
          Ingrendient 1
        </li>
        <li>
          Ingrendient 2
        </li>
      </ul>
    </div>
    <div>
      <h3>
        Steps
      </h3>
      <ol>
        <li>
          Steps 1
        </li>
        <li>
          Steps 2
        </li>
      </ol>
    </div>
    <a
      className="right-align"
      href="/recipe/1"
      onClick={[Function]}
    >
      Detail
    </a>
  </div>
</div>
`;

exports[`<RecipeDetail /> Should render zero state if no recipe passed 1`] = `
<p
  className="h3 p2 bg-white italic center"
  style={undefined}
>
  No recipe selected.
</p>
`;
